practice/jokes/jokes/logic.py

```diff
@@ -0,0 +1,21 @@
+"""jokes Logic."""
+
+from . import models
+
+
+def get_all_jokes():
+    """Returns all jokes in submitted order."""
+    return models.jokes
+
+
+def joke_is_valid(setup, punchline):
+    """Tests if both setup and punchline fields in submitted joke have a value."""
+    return len(setup) > 0 and len(punchline) > 0
+
+
+def add_new_joke(setup, punchline):
+    """If joke is valid, adds joke as a Joke class object to global variable containing all jokes."""
```

Comment:
Doctest. Try using the [functional core, state-ful shell](https://github.com/PdxCodeGuild/Full-Stack-Day-Class/blob/master/sessions/2016-07-07/tips/2016-08-31.md#functional-core-stateful-shell) trick to make it more testable.